src/vanilla/DanModel/JavaMethodGroupInterface.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace AutoRest.Java.DanModel
+{
+    class JavaMethodGroupInterface : JavaFileGenerator
+    {
+        public JavaMethodGroupInterface(IEnumerable<string> imports, string typeName, IEnumerable<JavaMethod> methods)
+        {
+            Imports = imports;
+            TypeName = typeName;
+            Methods = methods;
+        }
+
+        protected IEnumerable<string> Imports { get; }
+
+        protected string TypeName { get; }
+
+        protected IEnumerable<JavaMethod> Methods { get; }
+
+        protected override string FileNameWithoutExtension => TypeName;
+
+        public override JavaFile GenerateJavaFile(string folderPath, string headerComment, string package, int maximumHeaderCommentWidth)
+        {
+            return GenerateJavaFileWithHeaderAndPackage(folderPath, headerComment, package, maximumHeaderCommentWidth)
+                .Import(Imports)
+                .WordWrappedMultipleLineComment(maximumHeaderCommentWidth, (comment) =>
+                {
+                    comment.Line($"An instance of this class provides access to all the operations defined in {TypeName}.");
+                })
+                .PublicInterface(TypeName, (typeBlock) =>
+                {
+                    foreach (JavaMethod method in Methods)
+                    {
+                        typeBlock.MultipleLineComment((comment) =>
+                        {
+                            comment.Line(method.Description);
+                            comment.Line();
+                            if (method.Parameters != null)
```

Comment:
Would encourage requiring people to provide empty collections instead of null unless null is useful for representing a special case.

Reply:
There's no special case for null here. This is just preferring to handle null instead of crashing.